worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java

```diff
@@ -408,9 +409,17 @@ public long getPausePushDataTime() {
     return pausePushDataTime;
   }
 
-  private void appendPauseSpentTime() {
+  public long getPausePushDataAndReplicateTime() {
+    return pausePushDataAndReplicateTime;
+  }
+
+  private void appendPauseSpentTime(ServingState servingState) {
     long nextPauseStartTime = System.currentTimeMillis();
-    pausePushDataTime += nextPauseStartTime - pauseStartTime;
+    if (servingState == ServingState.PUSH_PAUSED) {
```

Comment:
Here is a logic problem. Sometimes the worker's serving state will change from NONE_PAUSED to PUSH_AND_REPLICATE_PAUSED if the worker's memory pressure is very high. This timer is incorrect if the scenerio happens.